### PR TITLE
disable language detection from browser when running tests

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

I wanted to run cypress on my local computer, but my system is in French.
So I can't pass the login test.

### CODE REVIEW

File: src/Session.php [hunk 1]
```diff
@@ -861,17 +861,19 @@ public static function getPreferredLanguage(): string
...
         $accepted_languages = [];
-        $values = explode(',', $_SERVER['HTTP_ACCEPT_LANGUAGE'] ?? '');
-        foreach ($values as $value) {
-            $parts = explode(';q=', trim($value));
-            $language = str_replace('-', '_', $parts[0]);
-            $qfactor  = $parts[1] ?? 1; //q-factor defaults to 1
-            $accepted_languages[$language] = $qfactor;
-        }
-        arsort($accepted_languages); // sort by qfactor
+        if (GLPI_ENVIRONMENT_TYPE !== 'testing') {
```
Comment (cedric-anne) [inline src/Session.php:867]: You can open this if block at line 863 and close it at line 883. There is no need to execute the `foreach (array_keys($accepted_languages) as $language) {` loop if you prevent anything to be added to the `$accepted_languages` array.

**Review by cedric-anne — COMMENTED** [review]
It will make this method untestable, and therefore break existing tests.

File: src/Session.php [hunk 2]
```diff
@@ -856,26 +856,27 @@ public function __construct(?I18nCache $cache)
...
+    public static function getPreferredLanguage(bool $force_headers_test = false): string
     {
         /** @var array $CFG_GLPI */
         global $CFG_GLPI;
 
-       // Extract accepted languages from headers
-       // Accept-Language: fr-FR, fr;q=0.9, en;q=0.8, de;q=0.7, *;q=0.5
+        // Extract accepted languages from headers
+        // Accept-Language: fr-FR, fr;q=0.9, en;q=0.8, de;q=0.7, *;q=0.5
         $accepted_languages = [];
```
Comment (cedric-anne) [inline src/Session.php:866]:
```suggestion
        if (GLPI_ENVIRONMENT_TYPE !== 'testing') {
            // Extract accepted languages from headers
            // Accept-Language: fr-FR, fr;q=0.9, en;q=0.8, de;q=0.7, *;q=0.5
            $accepted_languages = [];
```

File: src/Session.php [hunk 3]
```diff
@@ -856,26 +856,27 @@ public function __construct(?I18nCache $cache)
...
-            $language = str_replace('-', '_', $parts[0]);
-            $qfactor  = $parts[1] ?? 1; //q-factor defaults to 1
-            $accepted_languages[$language] = $qfactor;
-        }
-        arsort($accepted_languages); // sort by qfactor
-
-        foreach (array_keys($accepted_languages) as $language) {
-            if (array_key_exists($language, $CFG_GLPI['languages'])) {
-                return $language;
+        if (GLPI_ENVIRONMENT_TYPE !== 'testing') {
```
Comment (cedric-anne) [inline src/Session.php:867]:
```suggestion
```

**Review by cedric-anne — APPROVED** [review]

**Review by cconard96 — APPROVED** [review]

**Comment by cconard96** [comment]
An alternative fix is to add the Accept-Language header in Cypress when it visits the login page.
In the commands.js file for the login command:
```js
cy.visit('/', {
    headers: {
        'Accept-Language': 'en-GB,en;q=0.9',
    }
});
```

**Comment by orthagh** [comment]
> An alternative fix

I'll do a test, I did a lot of things in Cypress config and in the test file yesterday, no one worked.
I think I tested the Accept-Language header but I'm not 100% sure

**Comment by cedric-anne** [comment]
> An alternative fix is to add the Accept-Language header in Cypress when it visits the login page. In the commands.js file for the login command:
> 
> ```js
> cy.visit('/', {
>     headers: {
>         'Accept-Language': 'en-GB,en;q=0.9',
>     }
> });
> ```

It is probably a simplier and cleaner solution indeed.

**Comment by orthagh** [comment]
Seems to work.

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**